src/data/list/of_fn.lean

```diff
@@ -185,6 +185,10 @@ nat.rec_on n (by simp) $ λ n ihn, by simp [ihn]
 by simp_rw [of_fn_mul, ←of_fn_const, fin.repeat, fin.mod_nat, fin.coe_mk,
   add_comm, nat.add_mul_mod_self_right, nat.mod_eq_of_lt (fin.is_lt _), fin.eta]
 
+lemma _root_.list.pairwise_of_fn {R : α → α → Prop} {n} (f : fin n → α)
```

Comment:
You're already in the list namespace here
```suggestion
lemma pairwise_of_fn {R : α → α → Prop} {n} (f : fin n → α)
```

Reply:
I would prefer we split this off to a new PR so that we can more easily sync it with mathlib4. It would be good to have the `iff` version at the same time.

Reply:
What's your rationale against half underwater PRs? As far as I understood, they are fine.